tests/networkpolicy_test.go

```diff
@@ -1,173 +1,236 @@
+		tests.BeforeAll(func() {
+			tests.BeforeTestCleanup()
+
+			// Create three vmis, serverVMI and clientVMI are in same namespace, clientVMIAlternativeNamespace is in different namespace
+			serverVMI = createVMICirros(virtClient,
+				tests.NamespaceTestDefault,
+				serverVMILabels,
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(
+					v1.Port{
+						Name:     "http80",
+						Port:     80,
+						Protocol: "TCP",
+					},
+					v1.Port{
+						Name:     "http81",
+						Port:     81,
+						Protocol: "TCP",
+					},
+				)),
+			)
+			clientVMI = createVMICirros(virtClient, tests.NamespaceTestDefault, map[string]string{}, libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()))
+			clientVMIAlternativeNamespace = createVMICirros(virtClient, tests.NamespaceTestAlternative, map[string]string{}, libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()))
 
-	Context("vms limited by Default-deny networkpolicy", func() {
-		var policy *v1network.NetworkPolicy
-
-		BeforeEach(func() {
-			// deny-by-default networkpolicy will deny all the traffice to the vms in the namespace
-			By("Create deny-by-default networkpolicy")
-			networkpolicy := &v1network.NetworkPolicy{
-				ObjectMeta: v13.ObjectMeta{
-					Name: "deny-by-default",
-				},
-				Spec: v1network.NetworkPolicySpec{
-					PodSelector: v13.LabelSelector{},
-					Ingress:     []v1network.NetworkPolicyIngressRule{},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
-		})
+			serverVMI = tests.WaitUntilVMIReady(serverVMI, tests.LoggedInCirrosExpecter)
+			clientVMI = tests.WaitUntilVMIReady(clientVMI, tests.LoggedInCirrosExpecter)
+			clientVMIAlternativeNamespace = tests.WaitUntilVMIReady(clientVMIAlternativeNamespace, tests.LoggedInCirrosExpecter)
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+			By("Start HTTP server at serverVMI on ports 80 and 81")
+			tests.StartServer(serverVMI, tests.HTTPServer, 80)
+			tests.StartServer(serverVMI, tests.HTTPServer, 81)
+
+			assertIPsNotEmptyForVMI(serverVMI)
+			assertIPsNotEmptyForVMI(clientVMI)
+			assertIPsNotEmptyForVMI(clientVMIAlternativeNamespace)
 
-		It("[test_id:1511] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
 		})
 
-		It("[test_id:1512] should be failed to reach vmib from vmia", func() {
-			By("Connect vmib from vmia")
-			assertPingFail(vmia, vmib)
+		Context("and connectivity between VMI/s is blocked by Default-deny networkpolicy", func() {
+			var policy *networkv1.NetworkPolicy
+
+			BeforeEach(func() {
+				// deny-by-default networkpolicy will deny all the traffic to the vms in the namespace
+				policy = createNetworkPolicy(serverVMI.Namespace, "deny-by-default", metav1.LabelSelector{}, []networkv1.NetworkPolicyIngressRule{})
```

Comment:
maybe create a method createDenyByDefalutNetworkPolicy, or something like that,  to better the readability,
or you think this is redundant?

Reply:
the name of the profile describe it.